indico/modules/events/contributions/forms.py

```diff
@@ -22,18 +22,18 @@
 from wtforms.validators import DataRequired
 
 from indico.core.db.sqlalchemy.protection import ProtectionMode
-from indico.modules.events.contributions.fields import ContributionPersonListField
+from indico.modules.events.contributions.fields import ContributionPersonListField, SubContributionPersonListField
 from indico.web.forms.base import IndicoForm
 from indico.web.forms.fields import TimeDeltaField, PrincipalListField, IndicoEnumRadioField, IndicoLocationField
-from indico.web.forms.validators import UsedIf
+from indico.web.forms.validators import UsedIf, MaxDuration
 from indico.util.i18n import _
 
 
 class ContributionForm(IndicoForm):
     title = StringField(_("Title"), [DataRequired()])
     description = TextAreaField(_("Description"))
-    duration = TimeDeltaField(_("Duration"), [DataRequired()], default=timedelta(minutes=20),
-                              units=('minutes', 'hours'),
+    duration = TimeDeltaField(_("Duration"), [DataRequired(), MaxDuration(timedelta(hours=24))],
```

Comment:
I wouldn't say that either. One thing is a contribution being able to last until after midnight and another that they can span several days. 24 hours seem a reasonable max value. 

Contributions lasting longer than midnight don't look particularly pretty, but I guess it's a very rare case.